pkg/cli/config.go

```diff
@@ -95,20 +95,24 @@ func BaseConfig() *Config {
 	}
 }
 
-func LoadConfig(srcs []string, c *Config) error {
-	combined := []byte{}
-	for _, path := range srcs {
-		b, err := os.ReadFile(path)
-		if err != nil {
-			return fmt.Errorf("error reading config file: %w", err)
-		}
-		combined = append(combined, b...)
+func LoadConfig(sysSrc, userSrc string, c *Config) error {
+	// Load system configuration
+	if err := loadConfig(sysSrc, c); err != nil {
+		return fmt.Errorf("error loading system configuration: %w", err)
 	}
-	decoder := toml.NewDecoder(bytes.NewReader(combined))
-	// Fail if there is some unknown configuration. This is better than silently
-	// ignoring a (perhaps misspelled) config entry.
-	decoder.Strict(true)
-	return decoder.Decode(c)
+	if userSrc == "" {
+		return nil
+	}
+	sysSrvcs := c.Services
```

Comment:
While probably not a problem in practice, it's better to make a copy of the elements or set `c.Services` to `nil` after this assignment. `c` is passed by reference and is expected to be modified by the json package, which may or may not override this property. AFAIK it's not guaranteed that it will not reuse the underlying array in `c.Services`, of which `sysSrvcs` is also a reference of (a slice).

Reply:
True, I loaded the system config a second time earlier to have a deep copy of the system services and avoid using the `reflect` package. Went with `c.Services = nil` in this case.